src/nvim/eval.c

```diff
@@ -7586,16 +7586,20 @@ static void f_assert_equal(typval_T *argvars, typval_T *rettv)
 }
 
 // Common for assert_true() and assert_false().
-static void assert_bool(typval_T *argvars, bool isTrue)
+static void assert_bool(typval_T *argvars, bool is_true)
 {
   int error = (int)false;
   garray_T ga;
 
-  if (argvars[0].v_type != VAR_NUMBER ||
-      (get_tv_number_chk(&argvars[0], &error) == 0) == isTrue || error) {
+  if ((argvars[0].v_type != VAR_NUMBER ||
+       (get_tv_number_chk(&argvars[0], &error) == 0) == is_true || error)
+      && (argvars[0].v_type != VAR_SPECIAL
+          || argvars[0].vval.v_special != (is_true
+                                           ?kSpecialVarTrue
+                                           :kSpecialVarFalse))) {
```

Comment:
Help needs to be updated for using special values in `assert_{true,false}()`:

The help for `assert_false()` says `A value is false when it is zero. When "{actual}" is not a number the assert fails.`. This could mean a fail for all special values or, if the described conversion in the help for special values is used, that **`v:null`** and `v:false` should not fail, because they are converted to `0`.